ceph/rados/monitor_workflows.py

```diff
@@ -0,0 +1,260 @@
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mon service exists before removing
+        host = kwargs.get("host")
+        if self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon Already present on the host {host}. Returning Pass")
+            return True
+
+        log.debug(f"Adding mon daemon on the host : {host}")
+        cmd = f"ceph mon add {host} {kwargs['ip']}"
+        if kwargs.get("location_type", False):
+            cmd = cmd + f" {kwargs['location_type']}={kwargs['location_name']}"
+        log.debug(f"Adding new mon via cmd : {cmd}")
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Sleeping for 10 seconds for the mon to be added
+        time.sleep(10)
+        cmd = f"ceph orch daemon add mon {host}"
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Sleeping for 10 seconds for the mon to be added
+        time.sleep(10)
+
+        if not self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon not present on the host {host}. Returning Fail")
+            return False
+        log.info(f"Mon service running on host {host} post addition")
+
+        # Adding mon label on host if not present
+        labels = self.get_host_labels(host=host)
+        if "mon" not in labels:
+            cmd = f"ceph orch host label add {host} mon"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            log.debug(f"Added mon label from host : {host}")
+
+        log.debug(f"Mon successfully added on host : {host}")
+        return True
+
+    def get_mon_quorum_leader(self) -> str:
+        """
+        Fetches mon details and returns the name of quorum leader
+        Returns: name of the leader mon in quorum
+        """
+        cmd = "ceph mon stat"
+        quorum = self.rados_obj.run_ceph_command(cmd)
+        return quorum["leader"]
+
+    def get_mon_quorum_hosts(self) -> list:
+        """
+        Method to fetch the current hosts in mon quorum
+
+        returns:
+            List of hostnames that are present in the mon quorum
+        """
+        cmd = "ceph mon stat"
+        quorum = self.rados_obj.run_ceph_command(cmd)
+        return [entry["name"] for entry in quorum["quorum"]]
+
+    def set_tiebreaker_mon(self, host) -> bool:
+        """
+        Sets the passed host mon as the new tiebreaker mon daemon in stretch mode
+
+        Args:
+            host: name of the host which should be set as the tiebreaker mon
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        cmd = f"ceph mon set_new_tiebreaker {host}"
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Checking if the tiebreaker mon is added successfully
+        cmd = "ceph mon dump"
+        out = self.rados_obj.run_ceph_command(cmd=cmd, client_exec=True)
+        log.debug(f"The mon dump is : {out}")
```

Comment:
This could be a huge output, any chance we could avoid printing it?

Reply:
Or maybe let's keep it, shall help in debugging failures

Reply:
It is not huge, and it's good info to have during debugging.